tests/framework/clients/corral/corral.go

```diff
@@ -0,0 +1,137 @@
+package corral
+
+import (
+	"bytes"
+	"io"
+	"os/exec"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	debugFlag = "--debug"
+)
+
+// SetupCorralConfig sets the corral config vars. It takes a map[string]string as a parmeter; the key is the value and the value the value you are setting
+func SetupCorralConfig(configVars map[string]string) error {
+	for variable, value := range configVars {
+		msg, err := exec.Command("corral", "config", "vars", "set", variable, value).CombinedOutput()
+		if err != nil {
+			return errors.Wrap(err, "SetupCorralConfig: "+string(msg))
+		}
+	}
+	return nil
+}
+
+// CreateCorral creates a corral taking the corral name, the package path, and a debug set so if someone wants to view the
+// corral create log
+func CreateCorral(corralName, packagePath string, debug bool) (*rest.Config, error) {
+	if debug {
+		msg, err := exec.Command("corral", "create", corralName, packagePath, debugFlag).CombinedOutput()
+		if err != nil {
+			return nil, errors.Wrap(err, "CreateCorral: "+string(msg))
+		}
+
+		myString := string(msg[:])
+		logrus.Infof(myString)
+	} else {
+		msg, err := exec.Command("corral", "create", corralName, packagePath).CombinedOutput()
+		if err != nil {
+			return nil, errors.Wrap(err, "CreateCorral: "+string(msg))
+		}
+	}
+
+	kubeConfig, err := GetKubeConfig(corralName)
+	if err != nil {
+		return nil, err
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateK3DCluster: failed to parse kubeconfig for k3d cluster")
```

Comment:
Why is k3d mentioned here?

Reply:
Yeah this is a typo, k3d has nothing to do with this. I'll fix it